Chore: Changelog for Gateway 3.8 (#7857)
Kong/docs.konghq.com: app/gateway/changelog.md

See the [PostgreSQL settings section](/gateway/3.8.x/configuration/reference/#postgres-settings) in the Kong configuration reference for details.
* Added keyring encryption support to [license database entity payloads](/gateway/3.8.x/kong-enterprise/db-encryption/#configure-license-payload-encryption).
* Added support for a configurable STS endpoint for RDS IAM Authentication with the following new configuration fields: `pg_iam_auth_sts_endpoint_url` and `pg_ro_iam_auth_sts_endpoint_url`.
See the [PostgreSQL settings section](/gateway/3.8.x/configuration/reference/#postgres-settings) in the Kong configuration reference for details.
* Added support for a configurable STS endpoint for AWS Vault. This can either be configured by [`vault_aws_sts_endpoint_url`](/gateway/3.8.x/configuration/reference/#vault_aws_sts_endpoint_url) as a global configuration, or [`sts_endpoint_url`](/gateway/3.8.x/kong-enterprise/secrets-management/backends/aws-sm/) on a custom AWS vault entity.

#### Kong Manager

* Improved accessibility in Kong Manager.

>  [#13522](https://github.com/Kong/kong-manager/issues/13522)

Author: ```suggestion
```

* Enhanced entity lists so that you can resize or hide list columns.

>  [#13522](https://github.com/Kong/kong-manager/issues/13522)

Author: Nice catch, looks like another case of bad URL conversion from OSS.

>  [#13522](https://github.com/Kong/kong-manager/issues/13522)

Author: ```suggestion
```

* Added an SNIs field to the certificate form.

>  [#264](https://github.com/Kong/kong-manager/issues/264)

Author: ```suggestion
```

   [#13183](https://github.com/Kong/kong/issues/13183)

* [**AppDynamics**](/hub/kong-inc/app-dynamics/) (`app-dynamics`)
  * Added a new `ANALYTICS_ENABLE` flag. This plugin now also collects more snapshot user data in runtime.

* [**AWS Lambda**](/hub/kong-inc/aws-lambda) (`aws-lambda`)
  * Added support for a configurable STS endpoint with the new configuration field `aws_sts_endpoint_url`.
  [#13388](https://github.com/Kong/kong/issues/13388)
  * Added the configuration field `empty_arrays_mode` to control whether Kong should send `[]` empty arrays 
  (returned by Lambda function) as `[]` empty arrays or `{}` empty objects in JSON responses.
  [#13084](https://github.com/Kong/kong/issues/13084)

* [**JWT Signer**](/hub/kong-inc/jwt-signer/) (`jwt-signer`)
  * This plugin now supports using the `/jwt-signer/jwks/:jwt_signer_jwks` endpoint in DB-less mode.

* [**LDAP Authentication Advanced**](/hub/kong-inc/ldap-auth-advanced/) (`ldap-auth-advanced`)
  * This plugin now supports decoding an empty sequence or set represented in long form length.

* [**OpenID Connect**](/hub/kong-inc/openid-connect/) (`openid-connect`)

>   * Added support for Redis caching for introspection result with the new fields `cluster_cache_strategy` and `cluster_cache_redis`. 

Reviewer: ```suggestion
  * Added support for Redis caching introspection results with the new fields `cluster_cache_strategy` and `cluster_cache_redis`. 
```

It now controls the maximum time interval between two flushes of analytics messages to the configured backend, 
which means that if enough (less than `analytics_buffer_size_limit`) messages have already been buffered, 
the flush will happen before the configured interval. 
Previously, Kong always tried to flush messages after the configured interval, regardless of the number of 
messages in the buffer.
* Fixed an issue where `debug_listen` incorrectly used the SSL-related configuration of `status_listen`.

#### Core

* Fixed an issue where `luarocks-admin` was not available in `/usr/local/bin`.
 [#13372](https://github.com/Kong/kong/issues/13372)
* Fixed an issue where `read` was not always passed to PostgreSQL read-only database operations.
 [#13530](https://github.com/Kong/kong/issues/13530)
* Fixed the behavior of shorthand fields, which are used to describe deprecated fields:
  * Fixed an issue with deprecated shorthand fields so that they don't take precedence over replacement fields when both are specified.
  [#13486](https://github.com/Kong/kong/issues/13486)
  * Changed the way deprecated shorthand fields are used with new fields.

>   If the new field contains null, the deprecated field will overwrite it if both are present in the request.

Reviewer: ```suggestion
  If the new field contains `null`, the deprecated field will overwrite it if both are present in the request.
```

   [#13417](https://github.com/Kong/kong/issues/13417)

* [**ACME**](/hub/kong-inc/acme/) (`acme`) 
  * Fixed an issue where the DP would report that deprecated config fields were used when configuration was pushed from the CP.
   [#13069](https://github.com/Kong/kong/issues/13069)
  * Fixed an issue where username and password were not accepted as valid authentication methods.
   [#13496](https://github.com/Kong/kong/issues/13496)

* [**AWS Lambda**](/hub/kong-inc/aws-lambda) (`aws-lambda`)
  * Fixed an issue where the plugin didn't work with `multiValueHeaders` defined in proxy integration and legacy `empty_arrays_mode`.
   [#13381](https://github.com/Kong/kong/issues/13381)
  * Fixed an issue where the `version` field wasn't set in the request payload when `awsgateway_compatible` was enabled.
   [#13018](https://github.com/Kong/kong/issues/13018)

* [**CORS**](/hub/kong-inc/cors/) (`cors`)
  * Fixed an issue where the `Access-Control-Allow-Origin` header was not sent when `conf.origins` had multiple entries but included `*`.
   [#13334](https://github.com/Kong/kong/issues/13334)

* [**Correlation ID**](/hub/kong-inc/correlation-id/) (`correlation-id`)

>   * Fixed an issue where the plugin would not work if we explicitly set the `generator` to `null`.

Reviewer: ```suggestion
  * Fixed an issue where the plugin would not work if you explicitly set the `generator` to `null`.
```

* [**gRPC-Gateway**](/hub/kong-inc/grpc-gateway/) (`grpc-gateway`)
  * When there is a JSON decoding error, the plugin now responds with status 400 and error information in the body instead of status 500.
   [#12971](https://github.com/Kong/kong/issues/12971)

* [**HMAC Authentication**](/hub/kong-inc/hmac-auth/) (`hmac-auth`), JWT, LDAP Authentication, and OAuth2
  * Added WWW-Authenticate headers to 401 responses.
   [#11791](https://github.com/Kong/kong/issues/11791)
   [#11792](https://github.com/Kong/kong/issues/11792)
   [#11820](https://github.com/Kong/kong/issues/11820)
   [#11833](https://github.com/Kong/kong/issues/11833)

* [**HTTP Log**](/hub/kong-inc/http-log/) (`http-log`)
  * Fixed an issue where the plugin didn't include port information in the HTTP host header when sending requests to the log server.
   [#13116](https://github.com/Kong/kong/issues/13116)

* [**OAS Validation**](/hub/kong-inc/oas-validation/) (`oas-validation`)
  * Fixed an issue where the plugin couldn't obtain the value when the path parameter name contained hyphen characters.

* [**OpenTelemetry**](/hub/kong-inc/opentelemetry/) (`opentelemetry`)

>   * Fixed an issue where migration failed when upgrading from below version 3.3 to 3.7.

Reviewer: ```suggestion
  * Fixed an issue where migration failed when upgrading from versions earlier than 3.3.x to 3.7.x.
```

   [#13391](https://github.com/Kong/kong/issues/13391)
  * Removed redundant deprecation warnings.
   [#13220](https://github.com/Kong/kong/issues/13220)
  * Improved the accuracy of sampling decisions.
   [#13275](https://github.com/Kong/kong/issues/13275)

* [**Zipkin**](/hub/kong-inc/zipkin/) (`zipkin`)
  * Removed redundant deprecation warnings.
   [#13220](https://github.com/Kong/kong/issues/13220)
  * Improved the accuracy of sampling decisions.
   [#13275](https://github.com/Kong/kong/issues/13275)

* [**Request Transformer**](/hub/kong-inc/request-transformer/) (`request-transformer`)
  * Fixed an issue where renamed query parameters, url-encoded body parameters, 
  and JSON body parameters were not handled properly when the target name was the same as the source name in the request.
  [#13358](https://github.com/Kong/kong/issues/13358)

* [**Basic Auth**](/hub/kong-inc/basic-auth/) (`basic-auth`)

>   * Fixed an issue where the realm field wasn't recognized for older Kong Gateway versions (before 3.6).

Reviewer: ```suggestion
  * Fixed an issue where the realm field wasn't recognized for older Kong Gateway versions (earlier than 3.6.x).
```

   [#13042](https://github.com/Kong/kong/issues/13042)
   * Added WWW-Authenticate headers to all 401 responses and realm option.
   [#11833](https://github.com/Kong/kong/issues/11833)

* [**Key Auth**](/hub/kong-inc/key-auth/) (`key-auth`)

>   * Fixed an issue where the realm field wasn't recognized for older Kong Gateway versions (before 3.7).

Reviewer: ```suggestion
  * Fixed an issue where the realm field wasn't recognized for older Kong Gateway versions (earlier than 3.7).
```

    from the `window_size` in the default config, the rate limiting of that consumer group would fall back to local strategy.
  * Fixed an issue where the sync timer could stop working due to a race condition.

* [**Proxy Cache**](/hub/kong-inc/proxy-cache/) (`proxy-cache`)
  * Fixed an issue where the Age header was not being updated correctly when serving cached responses.
   [#13387](https://github.com/Kong/kong/issues/13387)

* [**OAuth 2.0 Introspection**](/hub/kong-inc/oauth2-introspection/) (`oauth2-introspection`)
  * Fixed an issue where the consumer's cache couldn't be invalidated when the OAuth2 Introspection plugin used `client_id` as `consumer_by`.

* [**OpenID Connect**](/hub/kong-inc/openid-connect/) (`openid-connect`)
  * Fixed an issue where anonymous consumers could be cached as nil under a certain condition.
  * Updated the rediscovery to use a short lifetime (5s) if the last discovery failed.
  * Fixed an issue where `using_pseudo_issuer` didn't work when sending `PATCH` request.

* [**AI Rate Limiting Advanced**](/hub/kong-inc/ai-rate-limiting-advanced/) (`ai-rate-limiting-advanced`)

>   * Edited the logic for the window adjustment and fixed missing passing window to shm.

Reviewer: ```suggestion
  * Edited the logic for the window adjustment and fixed missing passing window to shared memory.
```

* [**OAS Validation**](/hub/kong-inc/oas-validation/) (`oas-validation`)
  * Fixed an issue where parameter serialization didn't behave the same as in the OpenAPI specification.

>   * Fixed an issue where the non-string primitive types passed via URL query were unexpectedly cast to string when OpenAPI spec version was v3.1.0.

Reviewer: ```suggestion
  * Fixed an issue where the non-string primitive types passed via URL query were unexpectedly cast to string when the OpenAPI spec version was v3.1.0.
```

* Improved latency performance when gzipping/gunzipping large data (such as CP/DP config data).
 [#13338](https://github.com/Kong/kong/issues/13338)
* Improved the performance of Konnect Analytics by fetching the Rate Limiting context more efficiently.
* Improved the performance of Konnect Analytics by optimizing the buffering mechanism.

### Dependencies

- Bumped `lua-resty-acme` to 0.15.0 to support username/password auth with Redis.
 [#12909](https://github.com/Kong/kong/issues/12909)
- Bumped `lua-resty-aws` to 1.5.3 to fix a bug related to the STS regional endpoint.
 [#12846](https://github.com/Kong/kong/issues/12846)
- Bumped `lua-resty-events` to 0.3.0.
 [#13097](https://github.com/Kong/kong/issues/13097)
- Bumped `lua-resty-healthcheck` from 3.0.1 to 3.1.0 to reduce active healthcheck timer usage.
 [#13038](https://github.com/Kong/kong/issues/13038)
- Bumped `lua-resty-lmdb` to 1.4.3 (lmdb 0.9.33)
 [#12786](https://github.com/Kong/kong/issues/12786)
- Bumped `lua-resty-openssl` to 1.5.1.
 [#12665](https://github.com/Kong/kong/issues/12665)
- Bumped OpenResty to 1.25.3.2
 [#12327](https://github.com/Kong/kong/issues/12327)

> - Bumped PCRE2 to 10.44 to fix some bugs and tidy up the release.

Reviewer: ```suggestion
- Bumped PCRE2 to 10.44 to fix some bugs and organize the release.
```
